cmd/account/rotate-secret.go

```diff
@@ -80,128 +57,171 @@ func newRotateSecretOptions(streams genericclioptions.IOStreams, flags *genericc
+		return cmdutil.UsageErrorf(cmd, "Account CR argument is required")
 	}
-	o.iamUsername = args[0]
 
-	// account CR name and account ID cannot be empty at the same time
-	if o.accountName == "" && o.accountID == "" {
-		return cmdutil.UsageErrorf(cmd, "AWS account CR name and AWS account ID cannot be empty at the same time")
-	}
+	o.accountCRName = args[0]
 
-	if o.accountName != "" && o.accountID != "" {
-		return cmdutil.UsageErrorf(cmd, "AWS account CR name and AWS account ID cannot be set at the same time")
+	if o.profile == "" {
+		o.profile = "default"
 	}
 
-	// only initialize kubernetes client when account name is set
-	if o.accountName != "" {
-		var err error
-		o.kubeCli, err = k8s.NewClient(o.flags)
-		if err != nil {
-			return err
-		}
+	var err error
+	o.kubeCli, err = k8s.NewClient(o.flags)
+	if err != nil {
+		return err
 	}
 
 	return nil
 }
 
 func (o *rotateSecretOptions) run() error {
+
 	ctx := context.TODO()
 	var err error
-	awsSetupClient, err := awsprovider.NewAwsClient(o.profile, o.region, o.cfgFile)
+
+	// Get the associated Account CR from the provided name
+	var accountID string
+	account, err := k8s.GetAWSAccount(ctx, o.kubeCli, common.AWSAccountNamespace, o.accountCRName)
 	if err != nil {
 		return err
 	}
 
-	var accountID string
-	if o.accountName != "" {
-		account, err := k8s.GetAWSAccount(ctx, o.kubeCli, o.accountNamespace, o.accountName)
-		if err != nil {
-			return err
-		}
-		accountID = account.Spec.AwsAccountID
-	} else {
-		accountID = o.accountID
-	}
+	// Set the account ID
+	accountID = account.Spec.AwsAccountID
 
-	callerIdentityOutput, err := awsSetupClient.GetCallerIdentity(&sts.GetCallerIdentityInput{})
+	// Use provided profile
+	awsSetupClient, err := awsprovider.NewAwsClient(o.profile, "", "")
 	if err != nil {
 		return err
 	}
 
-	roleArn := aws.String(fmt.Sprintf("arn:aws:iam::%s:role/%s", accountID, awsv1alpha1.AccountOperatorIAMRole))
-	credentials, err := awsprovider.GetAssumeRoleCredentials(awsSetupClient, aws.Int64(900),
-		callerIdentityOutput.UserId, roleArn)
+	// Ensure AWS calls are succesful with client
+	callerIdentityOutput, err := awsSetupClient.GetCallerIdentity(&sts.GetCallerIdentityInput{})
 	if err != nil {
 		return err
 	}
 
+	var credentials *sts.Credentials
+	// Need to role chain if the cluster is CCS
+	if account.Spec.BYOC {
+		// TODO
+	} else {
+		roleArn := aws.String(fmt.Sprintf("arn:aws:iam::%s:role/%s", accountID, awsv1alpha1.AccountOperatorIAMRole))
+		credentials, err = awsprovider.GetAssumeRoleCredentials(awsSetupClient, aws.Int64(900),
+			callerIdentityOutput.UserId, roleArn)
+		if err != nil {
+			return err
+		}
+	}
+
 	awsClient, err := awsprovider.NewAwsClientWithInput(&awsprovider.AwsClientInput{
 		AccessKeyID:     *credentials.AccessKeyId,
 		SecretAccessKey: *credentials.SecretAccessKey,
 		SessionToken:    *credentials.SessionToken,
-		Region:          o.region,
+		Region:          "us-east-1",
```

Comment:
any reason why this is hard coded?

Reply:
Because region doesn't matter for IAM operations. No need to have a user provide one.